src/Workspaces/CSharp/Portable/Extensions/DirectiveSyntaxExtensions.DirectiveWalker.cs

```diff
@@ -148,6 +159,16 @@ private void HandleEndRegionDirective(DirectiveTriviaSyntax directive)
                 _directiveMap.Add(directive, previousDirective);
                 _directiveMap.Add(previousDirective, directive);
             }
+
+            internal void Finish() {
```

Comment:
This was to match VB behavior  It's also necessary so that we build up the 'conditional mapping' for if/else/elif directives when we don't see a ```#endif```